docs/keys_and_certificates.md

```diff
@@ -0,0 +1,263 @@
+- Fobnail Encryption key (encryption)
+    - Fobnail provisioning
+    - used for storage encryption
+
+### Platform
+
+#### Knows in advance
+
+- root CA for Platform Owner certificate chain
+    - remote platform provisioning
+    - local platform provisioning and attestation - Fobnail Identity key is signed
+    by Platform Owner, so Platform needs a way of verifying trust chain
+- Endorsement Key (encryption)
+    - remote and local platform provisioning
+    - for this project may be considered immutable persistent key unique to TPM
+- EK certificate
+    - EK is unique so is its certificate
+    - _usually_ saved in TPM NVRAM
+    - signed by TPM manufacturer CA
+
+#### Receives
+
+- Platform Owner certificate chain
+    - remote platform provisioning - directly from Platform Owner
+    - local platform provisioning and attestation - from Fobnail Token
+    - used for identification, TLS and verifying Fobnail Identity certificate
+- Fobnail Identity certificate
+    - local platform provisioning and attestation
+    - used for identification and TLS
+
+#### Produces
+
+- Attester Identity Key (signing)
+    - local and remote platform provisioning
+    - created by TPM
+    - trusted based on Make/ActivateCredential and pairing with EK instead of
+    certificate
+
+## Platform Owner certificate chain
+
+Platform owner certificate is used during Fobnail provisioning. Its root CA
+certificate must be preinstalled on Fobnail Token. Due to limited hardware
+capabilities of Fobnail Token, some restrictions apply.
+
+### Maximum certificate chain length
+
+Maximal supported certificate chain consists of 3 certificates: root CA (flashed
+to Fobnail's memory), intermediate CA and final CA that is used to issue
+certificates for keys generated by Fobnail. Intermediate CA is optional, root
+and issuing CA aren't.
+
+### File format
+
+Certificates are concatenated to a single PEM file. Order of concatenation
+matters, certificates must be written starting with leaf certificate and moving
+through intermediate towards root CA certificate. All of them must be present,
+despite the fact that root CA is already known by Fobnail.
+
+### Certificate extensions
+
+Trust chain will be traversed based on `X509v3 Authority Key Identifier`
+extension which is compared against `X509v3 Subject Key Identifier`, so both of
+them must be included in every certificate (former may be skipped for root CA).
+`X509v3 Basic Constraints` is required for CA certificates. `X509v3 Key Usage`
+must specify that all certificates in Platform Owner chain are able to sign
+certificates.
+
+### Validity checks
+
+Fobnail has neither timekeeping nor trusted networking capabilities. Because of
+that, validity period and certificate revocation cannot be tested. Only proper
+chain and attributes of certificates are validated.
+
+## Creating Platform Owner certificate chain with OpenSSL
+
+### Root CA
+
+Root CA certificate is self-signed. This means that signed public and signing
```

Comment:
Organizations with managed PKI may want to use one Root CA rooted deeper in some major authority e.g. VeriSign. There should be no difference in handling that, although typically in such scheme access to private key happen through API not directly.

Reply:
Yes, I wrote a paragraph about different possible use cases and somehow managed to lose it (probably got interrupted between cutting and pasting it when reordering the document). I think I mostly recreated it now few lines above.